src/SwedbankPay.Sdk.Infrastructure/PaymentOrders/OrderItem.cs

```diff
@@ -25,6 +26,7 @@ internal OrderItem(OrderItemDto dto)
             Description = dto.Description;
             DiscountDescription = dto.DiscountDescription;
             DiscountPrice = dto.DiscountPrice;
+            RestrictedToInstruments = dto.RestrictedToInstruments as IList<OrderItemInstrument>;

```

Comment:
Although this conversion through an implicit operator is clever, I like when things are more explicit. I would also like to have at least one test that ensures this conversion works in place.

Reply:
Thanks for your feedback. Before I go ahead and write test - would this be better, more explicit?
```cs
RestrictedToInstruments = dto.RestrictedToInstruments?.Select(instrument => OrderItemInstrument.FromValue(instrument)).ToList();
```

Reply:
Yes, that looks good to me. 👍🏼

Reply:
For test, I could write a method that:

1. Creates payment order with one order item with `RestrictedToInstruments` values. Will add a few values.
2. Checks the response if the same `RestrictedToInstruments` values can be found.

Would it be good?

Reply:
For test, I can write a method that:

1. Creates payment order with one order item with `RestrictedToInstruments` property. 
2. Checks the response order item if it has `RestrictedToInstruments` property with the same items.

If it passes, it means both request and response map correctly.

Reply:
Yep, that sounds good! 👍🏼

Reply:
I have updated this pull request with the changes.